inc/zoo/swar/SWAR.h

```diff
@@ -157,6 +164,58 @@ template<int NBits, typename T = uint64_t> struct SWAR {
     T m_v;
 };
 
+// SWAR is a useful abstraction for performing computations in lanes overlaid over any given integral type.
+// Doing additions via SWAR techniques requires an extra bit per lane be
+// available past the lane size, _or_ knowledge that both of your MSBs are set
+// 0 (leaving space for the operation).  Similarly, doing multiplications via
+// SWAR techniques require double bits per lane (unless you can bound your
+// inputs at half lane size)
+// This leads to a useful technique (which we use in the robin hood table)
+// where we interleave two related small bit count integers inside of a lane of
+// swar.  More generally, this is useful because it allows fast operations on
+// side a of a lane if side b is blitted out, and vice versa.  In the spirit of
+// separation of concerns, we provide a cut-lane-SWAR abstraction here.
+
+template<int NBitsSideM, int NBitsSideL, typename T = uint64_t> struct SWARWithSubLanes : SWAR<NBitsSideM+NBitsSideL, T> {
+  static constexpr inline auto Available = sizeof(T);
+  static constexpr inline auto LaneBits = NBitsSideL+NBitsSideM;
+  static constexpr inline auto NSlots = Available * 8 / LaneBits;
+
+  SWARWithSubLanes() = default;
+  constexpr explicit SWARWithSubLanes(T v): m_v(v) {}
+  constexpr explicit operator T() const noexcept { return m_v; }
+
+  // U is most significant bits slice, L is least significant bits slice.
+  // 0x....U2L2U1L1 or UN|LN||...||U2|L2||U1|L1
+  using SD = swar::SWAR<LaneBits, T>;
+  SD m_v;
+
+  //constexpr T Ones = makeBitmask<NBits, T>(SWAR<NBits, T>{1}.value());
+  static constexpr inline auto SideLOnes = makeBitmask<LaneBits, T>(SD{1}.value());
+  static constexpr inline auto SideMOnes = makeBitmask<LaneBits, T>(SD{1<<NBitsSideL}.value());
+  static constexpr inline auto SideLMask = makeBitmask<LaneBits, T>(SD{~(~0ull<<NBitsSideL)}.value());
+  static constexpr inline auto SideMMask = makeBitmask<LaneBits, T>(SD{SideLMask^(~(~0ull<<LaneBits))}.value());
+
+  constexpr auto sideL() const {
+    return m_v & SideLMask;
+  }
+  constexpr auto sideM() const {
+    return m_v & SideMMask;
+  }
+
+  // Sets the lsb sublane 
+  constexpr auto setSideL(T in, u8 pos) const {
+    return m_v.set( pos, (m_v.at(pos) & SideMMask) | in);
+  }
+
+  constexpr auto setSideM(T in, u8 pos) const {
```

Comment:
please drop `set`.  It adds nothing, the context of the overloads makes it clear they are a setter.  `set` and `get` **dilute** meaning.
Please watch this capital presentation, at least from the time mark: https://youtu.be/ZsHMHukIlJY?t=2361, Kevlin Henney's "Seven Ineffective..."